### DISCUSSION TITLE
What does the custom "Cast" menu in the expanded media controls do?

### DISCUSSION THREAD

**Discussion by EyMaddis** [react-native-google-cast/react-native-google-cast#292] [Q&A]
I see that on the device, but pressing it does nothing

https://github.com/react-native-google-cast/react-native-google-cast/blob/2e61c535b064b6b5ddc4370b886e43e0d5a9950c/android/src/main/java/com/reactnative/googlecast/RNGCExpandedControllerActivity.java#L18-L24

**Comment by petrbela** [comment]
That should render the Cast button in the expanded controller though tbh the expanded+mini controllers still need a bit of work so I wouldn't be totally surprised if it's misbehaving.

**Comment by arsimr16** [comment]
I see that the `cast_expanded_controller_menu.xml` file that was added in [this pull request](https://github.com/react-native-google-cast/react-native-google-cast/pull/78/files) is missing.  I thought maybe adding that back (or something like the code snippet below) might work, but I haven't had any luck so far.  It seems like the menu gets rendered but there's something missing that would either show the cast button here or trigger the same functionality as pressing the cast button.  It seems like the code matches the documentation that appears when I hover over `ExpandedControllerActivity`, so I don't really know what else to try.

```Define cast button menu xml
Developers need to define a menu xml that contains a cast button for the expanded controller Activity:
 // cast_expanded_controller_menu.xml
 <menu xmlns:android="//schemas.android.com/apk/res/android"
       xmlns:app="//schemas.android.com/apk/res-auto" >
     <item
     android:id="@+id/media_route_menu_item"
     app:actionProviderClass="androidx.mediarouter.app.MediaRouteActionProvider"
     app:showAsAction="always" />
 </menu>
 
Set up the cast button
Developers need to subclass ExpandedControllerActivity to set up the Cast button, and register the Activity in the application's manifest file:
 // MyExpandedControllerActivity.java
 class MyExpandedControllerActivity extends ExpandedControllerActivity {
     @Override
     public boolean onCreateOptionsMenu(Menu menu) {
         super.onCreateOptionsMenu(menu);
         getMenuInflater().inflate(R.menu.cast_expanded_controller_menu, menu);
         CastButtonFactory.setUpMediaRouteButton(this, menu, R.id.media_route_menu_item);
         return true;
     }
 }
```

I wish I could be more helpful, but I'm still very new to native Android.  I was hoping I could fix this through trial and error, but I think I need to deepen my Android knowledge because I'm not getting anywhere.  For now I think I'll just create a patch to remove the menu since it's weird having it there if the button doesn't do anything.
